docs/source/spec-workload-executor.rst

```diff
@@ -43,29 +43,98 @@ After accepting the inputs, the workload executor:
-     operations. An operation failure is when the actual return value of an operation does not match its
-     expected return value (as defined in the ``result`` field of the ``driverWorkload``).
-   * MUST keep count of the number of operation errors (``numErrors``) that are encountered while running
-     operations. An operation error is when running an operation unexpectedly raises an error. Workload executors
-     implementations should try to be as resilient as possible to these kinds of operation errors.
-   * MUST keep count of the number of operations that are run successfully (``numSuccesses``).
+     
+#. MUST use the driver's unified test runner to execute the tests in the
+   scenario, and the operations in each test, sequentially and in the order
+   in which they appear in the ``tests`` and ``operations`` arrays,
+   with the following deviations from the unified test runner specification:
+   
+   * The workload executor MUST repeat execution of the entire set of
+     specified tests and operations indefinitely, until the
+     **termination signal** from ``astrolabe`` is received.
+   
+   * The workload executor MUST keep count of the number of failures
+     (``numFailures``) that are encountered. A failure is when
+     the actual return value of an operation or a published event does not match
+     the respective expected return value or event (as these expectations
+     are defined in the unified test format).
+   
+   * The workload executor MUST record all errors encountered while running the scenario.
+     An operation error is any error that is propagated out of the unified test runner.
+     Workload executor implementations should try to be as resilient
+     as possible to these kinds of operation errors.
+   
+   * The workload executor MUST keep count of the number of operation errors (``numErrors``) that
+     are encountered while running the scenario.
+   
+   * The workload executor MUST keep count of the number of invocations of the scenario that
+     did not result in an error (``numSuccesses``).
 
 #. MUST set a signal handler for handling the termination signal that is sent by ``astrolabe``. The termination signal
    is used by ``astrolabe`` to communicate to the workload executor that it should stop running operations. Upon
    receiving the termination signal, the workload executor:
 
    * MUST stop running driver operations and exit soon.
-   * MUST dump collected workload statistics as a JSON file named ``results.json`` in the current working directory
+   * MUST write the collected events and errors into a JSON file named
+     ``events.json`` in the current directory
+     (i.e. the directory from where the workload executor is being executed). 
+     The data written MUST be a map with the following fields:
+     
+     * ``commands``: an array of command events published during scenario
```

Comment:
Can you modify this to say that this is an array of CommandSuccededEvent and CommandFailedEvents instead of simply calling them 'command events'?

Reply:
The structure of the data does not map to either CommandSuccededEvent or CommandFailedEvent, which is why I used the lowercase "command events" to avoid an expectation of a specific layout defined elsewhere. I am open to a different language but I think simply saying "CommandSuccededEvent or CommandFailedEvent" would not actually add clarity.

Reply:
Fair enough. Let's leave it as-is and we can clarify in a new PR if language teams complain when writing their integrations.